src/fi/fi.js

```diff
@@ -0,0 +1,394 @@
+    {
+        let target;
+        let ast = PARSER.getAST(id, target = 'idSimple');
+
+        if (ast.type !== 'idSimple' || ast.errors.length !== 0)
+        {
+            throw Error('Parameter id not a valid BasicHKID');
+        }
+        super(id);
+    }
+
+    toStringDecorated()
+    {
+        return super.toString();
+    }
+
+}
+
+class ExtendedHKID extends HKID
+{
+
+    constructor(id)
+    {
+        let target;
+        let ast = PARSER.getAST(`\`\`${id}\`\``, target = 'idExtended');
+
+        if (ast.type !== 'idExtended' || ast.errors.length !== 0)
+        {
+            throw Error('Parameter id not a valid ExtendedHKID');
+        }
+        super(id);
+    }
+
+    toStringDecorated()
+    {
+        return `\`\`${super.toString()}\`\``;
+    }
+
+}
+
+class IriHKID extends HKID
+{
+
+    constructor(id)
+    {
+        let target;
+        let ast = PARSER.getAST(`<${id}>`, target = 'iri');
+
+        if (ast.type !== 'iri' || ast.errors.length !== 0)
+        {
+            throw Error('Parameter id not a valid IriHKID');
+        }
+        super(id);
+    }
+
+    toStringDecorated()
+    {
+        return `<${super.toString()}>`;
+    }
+}
+
+
+//Try to emulate lark idiom for parsing
+
+function postProcessTree(ast)
+{
+    return processFijs(ast);
+}
+
+function processFijs(ast)
+{
+
+    if (ast.children.length === 1)
+    {
+        let artifact = processId(ast.children[0]);
+        return new FI(artifact);
+    } else if (ast.children.length >= 2)
+    {
+        let firstArtifact = processId(ast.children[0]);
+        let anchor = processAnchor(ast.children[1]);
+
+        let lastFI = new FI(firstArtifact, anchor)
+        for (let i = 2; i < ast.children.length; i++)
+        { //start on the 2nd anchor, if any
+            anchor = processAnchor(ast.children[i]);
+            lastFI = new FI(lastFI, anchor);
+        }
+
+        return lastFI;
+    } else
+    {
+        throw Error(`Error parsing ${ast.text}`)
+    }
+
+}
+
+
+function processAnchor(ast)
+{
+
+    if (ast.children.length == 1)
+    {
+        let indexer = processIndexer(ast.children[0]);
+        return new FIAnchor(indexer);
+    } else if (ast.children.length == 2)
+    {
+        let indexer = processIndexer(ast.children[0]);
+        if (ast.children[1].type === 'operator')
+        {
+            return new FIAnchor(indexer, null, FIOperator.DESCRIPTION);
+        } else
+        {
+            let token = processToken(ast.children[1]);
+            return new FIAnchor(indexer, token);
+        }
+    } else if (ast.children.length == 3)
+    {
+        let indexer = processIndexer(ast.children[0]);
+        let token = processToken(ast.children[2]);
+        return new FIAnchor(indexer, token, FIOperator.DESCRIPTION);
+    } else
+    {
+        throw Error(`Error parsing ${ast.text}`);
+    }
+
+}
+
+function processToken(ast)
+{
+    return processJsonPlus(ast.children[0])
+}
+
+function processJsonPlus(ast)
```

Comment:
Is JsonPlus a custom extended Json?

Reply:
It is not quite json. There is an special treatment of fijs strings (i.e. first case condition).